### ISSUE TITLE
Replace has-visible-text check with generic has-text-content

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#2194]
This would replace the has-visible-text and button-has-visible-text checks with a single check that will look something like this:

```js
return text.sanitize(axe.commons.text.subtreeText(virtualNode)) !== ""
```

For some rules (heading-empty) this will require adding other checks to the rule.any prop.

**Comment by padmavemulapati** [comment]
Not seeing changes at present, will verify it once the code changes appeared.

**Comment by padmavemulapati** [comment]
empty-heading test file 
`<!DOCTYPE html>
<html lang="en">
	<head>
		<title>empty-heading test</title>
		<meta charset="utf8" />
		<script src="/axe.js"></script>
	</head>
	<body>
		<h1 id="ignore1" role="banner">Meh</h1>

		<h1 id="pass1">Ok</h1>
		<h2 id="pass2">Ok</h2>
		<h3 id="pass3">Ok</h3>
		<h4 id="pass4">Ok</h4>
		<h5 id="pass5">Ok</h5>
		<h6 id="pass6">Ok</h6>
		<div id="pass7" role="heading">Ok</div>
		<h1 id="fail1"></h1>
		<h1 id="fail2"><div style="display: none">Not Ok</div></h1>
		<h1 id="fail3"></h1>
	</body>
</html>` previously it has the impact - 'serious', now as per the changes , its - 'minor'

![image](https://user-images.githubusercontent.com/47409579/84398278-64006d00-ac1d-11ea-837a-351eaccd8b80.png)
